src/index.ts

```diff
@@ -0,0 +1,104 @@
+import express from 'express';
+import { readdir } from 'fs/promises'
+import path from 'path';
+import helmet from 'helmet';
+import { cwd } from 'process';
+import { extname, join } from 'upath'
+
+import 'dotenv/config'
+
+
+let app = express();
+const PORT = 3000;
+
+app.use(helmet())
+app.use(express.json())
+
+
+async function getDirectories(directory_path: string) {
+    const directories = (await readdir(path.resolve(directory_path), {
+        withFileTypes: true
+    }))
+        .filter(dirent => dirent.isDirectory())
+        .map(dirent => dirent.name);
+
+    return directories;
+}
+
+async function getFiles(directory_path: string) {
+    const files = (await readdir(path.resolve(directory_path), {
+        withFileTypes: true
+    }))
+        .filter(dirent => dirent.isFile())
+        .map(dirent => dirent.name);
+
+    return files;
+}
+
+function getDirectoryPath(path: string) {
+    return process.env.ROOT_DIR === path ? process.env.ROOT_DIR : cwd();
+}
+
+//change to correct base directory
+const rootDirectory = '/c/User/david/workspace///';
+
+app.get('/', (_, res) => {
+    res.send({
+        1: '/api/fs/directory',
+        2: '/api/fs/directory/:directoryId/files'
+    })
+})
+
+//criteria 1
+app.get('/api/fs/directory', async (req, res) => {
+    //todo: make cross platform;
+
+    try {
+
+        const directory_path = getDirectoryPath(rootDirectory);
+
+        const files = await getDirectories(directory_path);
+        return res.send({ "directory": files })
+
+    } catch (err) {
+        console.log(err)
```

Comment:
when you get an unhandled error, you usually return a 500.  So here something like
```suggestion
        console.error(err)
        res.status(500).send()
```